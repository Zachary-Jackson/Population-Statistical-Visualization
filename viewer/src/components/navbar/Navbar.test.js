import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";

import Navbar from "./Navbar";

describe("Navbar", () => {
  const props = {
    items: [{ obj1: "cow" }, { obj2: "pig" }],
    onSelectPopulation: function() {}
  };

  it("renders without crashing", () => {
    const div = document.createElement("div");

    ReactDOM.render(<Navbar {...props} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test("has a valid snapshot", () => {
    const component = renderer.create(<Navbar {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
