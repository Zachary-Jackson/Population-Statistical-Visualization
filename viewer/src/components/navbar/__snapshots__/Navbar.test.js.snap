// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar has a valid snapshot 1`] = `
<div>
  <nav
    className="navbar navbar-dark bg-dark"
  >
    <a
      className="navbar-brand text-white"
      href=""
    >
      Population Statistical Visualization
    </a>
    <form
      className="form-inline my-2 my-lg-0"
      onSubmit={[Function]}
    >
      <input
        aria-label="Search"
        className="form-control mr-sm-2"
        name="item"
        onChange={[Function]}
        placeholder="Search populations"
        type="search"
        value=""
      />
      <button
        className="btn btn-success my-2 my-sm-0 mr-2"
        disabled={true}
        type="submit"
      >
        Search
      </button>
      <button
        className="btn btn-primary my-2 my-sm-0"
        onClick={[Function]}
        type="submit"
      >
        Random
      </button>
    </form>
  </nav>
</div>
`;
